source/neuropod/bindings/java/BUILD

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2020 UATC, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load(":java_build_defs.bzl", "JAVACOPTS")
+
+
+genrule(
+    name = "copy_link_jni_md_header",
+    srcs = select({
+        "@bazel_tools//src/conditions:darwin": ["@bazel_tools//tools/jdk:jni_md_header-darwin"],
+        "//conditions:default": ["@bazel_tools//tools/jdk:jni_md_header-linux"],
+    }),
+    outs = ["jni_md.h"],
+    cmd = "cp -f $< $@",
+)
+
+genrule(
+    name = "copy_link_jni_header",
+    srcs = ["@bazel_tools//tools/jdk:jni_header"],
+    outs = ["jni.h"],
+    cmd = "cp -f $< $@",
+)
+
+filegroup(
+    name = "native_libraries",
+    srcs = [
+      ":libneuropod_jni.so",
+      "//neuropod:libneuropod.so",
+      "//neuropod/multiprocess:neuropod_multiprocess_worker",
+      "//neuropod/backends/tensorflow:libneuropod_tensorflow_backend.so",
+      "//neuropod/backends/python_bridge:libneuropod_pythonbridge_backend.so",
+      "//neuropod/backends/torchscript:libneuropod_torchscript_backend.so",
+      "@tensorflow_repo//:libtensorflow_libs",
+      "@libtorch_repo//:libtorch_libs",
+    ] + select({
+        "@bazel_tools//src/conditions:darwin": ["@mklml_repo_darwin//:mklml_libs"],
+        "//conditions:default": [],
+        }),
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+)
+
+filegroup(
+    name = "java_source",
+    srcs = glob(["src/main/java/com/uber/neuropod/*.java"]),
+    visibility = [
+        "//neuropod:__subpackages__",
+    ],
+)
+
+java_binary(
+    name = "neuropod_java_fat_jar",
+    srcs = [":java_source"],
+    # TODO: Find a way to make these native libraries locate at /com/uber/neuropod/native/{OS}/{PLATFORM}
+    # in the jar file, instead of locating at the class root path.
+    classpath_resources = [":native_libraries"],
```

Comment:
Should it be runtime_deps instead https://docs.bazel.build/versions/master/be/java.html#java_binary.runtime_deps ?

Reply:
I use classpath_resources here because I want to test whether the Library loader can use the native libraries packed in the jar file. I think the fat jar will be used more widely because user does not need to install any external dependencies outside the java environment.